src/integrationTest/java/org/opensearch/security/http/OnBehalfOfJwtAuthenticationTest.java

```diff
@@ -141,6 +157,27 @@ public void shouldNotAuthenticateForNonAdminUserWithoutOBOPermission() {
         }
     }
 
+    @Test
+    public void shouldNotIncludeHostMappingInOBOToken() {
+        String oboToken = generateOboToken(OBO_USER_NAME_WITH_HOST_MAPPING, DEFAULT_PASSWORD);
+
+        String[] splitToken = oboToken.split("\\.");
+
+        String unsignedToken = splitToken[0] + "." + splitToken[1] + ".";
+        Claims claims = Jwts.parserBuilder().build().parseClaimsJwt(unsignedToken).getBody();
```

Comment:
Is splitting the token necessary for calling `parseClaimsJwt`?

Reply:
Yes, otherwise it won't recognize the verify signature part of the token, and throw an error:
```
Signed JWSs are not supported.
io.jsonwebtoken.UnsupportedJwtException: Signed JWSs are not supported.
...
```

Reply:
A couple lines down you are creating an instance of `EncryptionDecryptionUtil` with the expected encryption key. Can you do something similar with the JWTParser to avoid having to split the token?

Reply:
That's a good point. I just updated.